src/app/public/modules/description-list/description-list.component.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChildren,
+  ElementRef,
+  HostListener,
+  Input,
+  OnDestroy,
+  QueryList,
+  ViewChild
+} from '@angular/core';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  SkyDescriptionListMode
+} from './types/description-list-mode';
+
+import {
+  SkyDescriptionListOrientation
+} from './types/description-list-orientation';
+
+import {
+  SkyDescriptionListAdapterService
+} from './description-list-adapter-service';
+
+import {
+  SkyDescriptionListContentComponent
+} from './description-list-content.component';
+
+import {
+  SkyDescriptionListService
+} from './description-list.service';
+
+/**
+ * Creates a description list to display term-description pairs.
+ */
+@Component({
+  selector: 'sky-description-list',
+  templateUrl: './description-list.component.html',
+  styleUrls: ['./description-list.component.scss'],
+  providers: [SkyDescriptionListService],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyDescriptionListComponent implements AfterContentInit, OnDestroy {
+
+/**
+ * Specifies a default description to display when no description is provided
+ * for a term-description pair.
+ * @default 'None found'
+ */
+  @Input()
+  public set defaultDescription(value: string) {
+    this.descriptionListService.defaultDescription.next(value);
```

Comment:
Curious, did you investigate a way to implement the default description without an observable stream? Observables tend toward complexity, if we can avoid it.

Reply:
Well, I could always loop through the children components when `defaultDescription`, but that seems a lot less efficient, especially because I have 2 nested levels of children/grandchildren to go through (`description-list-content` > `description-list-description`). I also tried to get at the variable from the grandchild by using `@Optional() @Host()` injectors, but I think its not working b/c it's too many levels deep. This seems a lot cleaner and simpler than the alternatives. Plus, if the consumer pushes an update to the variable it will trickle down nicely due to the observable.